### ISSUE TITLE
google_checks.xml: false negatives in type variable names

### ISSUE THREAD

**Issue by Atharv3221** [checkstyle/checkstyle#20563]
I have read check documentation: 
https://checkstyle.sourceforge.io/checks/naming/methodtypeparametername.html, 
https://checkstyle.sourceforge.io/checks/naming/classtypeparametername.html, 
https://checkstyle.sourceforge.io/checks/naming/interfacetypeparametername.html,
https://checkstyle.sourceforge.io/checks/naming/recordtypeparametername.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

[5.2.8 Type variable names](https://google.github.io/styleguide/javaguide.html#s5.2.8-type-variable-names)
Each type variable is named in one of two styles:

A single capital letter, optionally followed by a single numeral (such as E, T, X, T2)
A name in the form used for classes (see Section 5.2.2, [Class names](https://google.github.io/styleguide/javaguide.html#s5.2.2-class-names)), followed by the capital letter T (examples: RequestT, FooBarT).

```java
atharv@atharv-IdeaPad-3-15IIL05:~/test$ cat -n Test.java
     1  /** Some javadoc. */
     2  public class Test<KEYT, VALUET> {}               // expected violation
     3
     4  /** Some javadoc. */
     5  record Foo<FOOT>() {}                            // expected violation
     6
     7  /** Some javadoc. */
     8  class Test2<T> {}                                // ok
```
```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ java -jar checkstyle-13.7.0-all.jar -c /google_checks.xml Test.java
Starting audit...
[WARN] /home/atharv/test/Test.java:5:1: Top-level class Foo has to reside in its own source file. [OneTopLevelClass]
[WARN] /home/atharv/test/Test.java:8:1: Top-level class Test2 has to reside in its own source file. [OneTopLevelClass]
Audit done.
```
Expected violations for Type variable names `KEYT`, `VALUET`, `FOOT`

As a fix google_checks.xml config regex needs to improved for checks at https://checkstyle.sourceforge.io/google_style.html#a5.2.8
